### PR TITLE
fix(st-dma2d): remove unused functions

### LINKED ISSUE #5542: v8.3 unused DMA STM32 functions
### Introduce the problem

There is a pending PR on our lvgl module which tries to improve the STM32 performance of LVGL. [https://github.com/zephyrproject-rtos/zephyr/pull/68254](https://github.com/zephyrproject-rtos/zephyr/pull/68254)
Everything works but CI is failing because there are 4 unused functions. 

```
_lv_gpu_stm32_dma2d_invalidate_cache
_lv_draw_stm32_dma2d_copy_buffer
lv_gpu_stm32_dma2d_wait_cb
lv_draw_stm32_dma2d_img
```

### Proposal

Either removal of the functions or introduction of an `LV_ATTRIBUTE_MAYBE_UNUSED`. 
I'd be happy to submit a PR for this.

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

fixes #5542

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by C47D — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Waiting for @faxe1008's  confirmation for merging. 

I'll solve the code formatting issue in an other PR.

File: src/draw/stm32_dma2d/lv_gpu_stm32_dma2d.h [hunk 1]
```diff
@@ -53,6 +57,16 @@ struct _lv_disp_drv_t;
 void lv_draw_stm32_dma2d_init(void);
 void lv_draw_stm32_dma2d_ctx_init(struct _lv_disp_drv_t * drv, lv_draw_ctx_t * draw_ctx);
 void lv_draw_stm32_dma2d_ctx_deinit(struct _lv_disp_drv_t * drv, lv_draw_ctx_t * draw_ctx);
+void _lv_draw_stm32_dma2d_copy_buffer(const lv_color_t * dest_buf, lv_coord_t dest_stride,
+                                                            const lv_area_t * draw_area, const lv_color_t * src_buf, lv_coord_t src_stride, const lv_point_t * src_offset);;
```
Comment (lhdjply) [inline src/draw/stm32_dma2d/lv_gpu_stm32_dma2d.h]: There's an extra semicolon.

Comment (kisvegabor) [reply src/draw/stm32_dma2d/lv_gpu_stm32_dma2d.h]: Updated, thank you!

**Review by lhdjply — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]
With this patch the module builds now without warnings, thanks @kisvegabor :+1:

**Comment by kisvegabor** [comment]
Great, merging!

**Pull request merged**